CONTRIBUTING.md

```diff
@@ -0,0 +1,68 @@
+# Contributing to the Self Hosted Show Wiki
+
+
+## Objectives of This Wiki
+
+In order to contribute well, you need to understand what the objectives are behind this wiki.
+
+What we are trying to create:
+- A compilation of tools, tips and tricks for self-hosting specific applications.
+- A place where beginners and experts can learn about how to quickly, efficiently and securely deploy applications on their own server(s).
+- Snippets that people can quickly reference in deploying applications.
+- A hub for quality documentation (to the extent possible).
+
+What we are not trying to do:
+- Be another large list of self-hosted tools. See [awesome-selfhosted](https://github.com/awesome-selfhosted/awesome-selfhosted) for a comprehensive list of self-hosted applications.
+- Be a hub for script kitties. People should _learn_ from documentation here not just come here to copy/paste. Snippets are important but should be documented properly.
+- Insecure, sloppy or any undesirable systems administration practices such as:
+  - Executing remote code by `curl`-ing an untrusted script and piping it to `sudo bash`: `curl https://example.com/awesomescript.sh | sudo bash`
+  - Installing additional bloat within a Docker container
+  - Etc.
+
+
+We want to encourage the use of dev/ops tools such as Docker, Ansible, SALT Stack, etc. for increased ease of use and for automating/standardizing as much as possible.
+
+
+## Directory Layout:
+
+The `docs` folder in the root of the repository is where wiki content should be placed.
+
+The following example directory layout should be adhered to when contributing:
+
+`docs/category/application/application.md`
+
+
+For example, if I wanted to add an article for Home Assistant, my layout could be this:
+
+`docs/home-automation/home-assistant/home-assistant.md`
```

Comment:
let's change the name from `home-assistant.md` to something less generic (if it isn't index.md). I fear that this will lead to folks creating dirs and just filling the dir full of files the same name as the project when what we should be aiming for is a commit message style naming structure. e.g. When I read this document I will learn how to 'deploy-home-assistant.md'.

Reply:
@IronicBadger as a compromise, this is what I put in CONTRIBUTING.md:

```
The following example directory layout should be adhered to when contributing:

docs/category/application/index.md

Additional, more specific articles relating to that same topic should be named differently:

docs/category/application/specific-topic.md

For example, if I wanted to add a brand new, generic article for Home Assistant, my layout would be this:

docs/home-automation/home-assistant/index.md

For an additional article that is more specific I could name it like so:

docs/home-automation/home-assistant/docker-deployment.md
```


Let me know if that works for you.